src/main/java/org/candlepin/subscriptions/security/SecurityConfig.java

```diff
@@ -42,144 +44,195 @@
+                    .antMatchers(String.format("/%s/ingress/**", apiPath)).permitAll()
+                    .requestMatchers(EndpointRequest.toAnyEndpoint()).permitAll()
+                    .anyRequest().authenticated();
+        }
     }
 
-    @Bean
-    public AccessDeniedHandler restAccessDeniedHandler() {
-        return new RestAccessDeniedHandler(mapper);
+    /**
+     * Security configuration for Jolokia Actuator.  Jolokia has GET endpoints that can affect the state of
+     * the application.  We need to protect these endpoints from CSRF attacks.  Normally GET requests are
+     * exempted from CSRF restrictions, so we need to create a special security configuration and use a
+     * special AntiCsrfFilter to guard GET requests on the /actuator/jolokia context.
+     *
+     * This security configuration will be loaded before the normal configuration.
+     */
+    @Order(1)
+    @Configuration
+    @Import(ApiConfiguration.class)
+    public static class JolokiaActuatorConfiguration extends WebSecurityConfigurerAdapter {
+        @Autowired private ApplicationProperties appProps;
+        @Autowired private ConfigurableEnvironment env;
+        @Autowired protected ObjectMapper mapper;
+
+        // NOTE: intentionally *not* annotated with @Bean; @Bean causes an extra use as an application filter
+        public AntiCsrfFilter getVerbIncludingAntiCsrfFilter(ApplicationProperties appProps,
+            ConfigurableEnvironment env) {
+            return new GetVerbIncludingAntiCsrfFilter(appProps, env);
+        }
+
+
+        // NOTE: intentionally *not* annotated w/ @Bean; @Bean causes an *extra* use as an application filter
+        public IdentityHeaderAuthenticationFilter identityHeaderAuthenticationFilter() throws Exception {
+            IdentityHeaderAuthenticationFilter filter = new IdentityHeaderAuthenticationFilter(mapper);
+            filter.setCheckForPrincipalChanges(true);
+            filter.setAuthenticationManager(authenticationManager());
+            filter.setAuthenticationFailureHandler(new IdentityHeaderAuthenticationFailureHandler(mapper));
+            filter.setContinueFilterChainOnUnsuccessfulAuthentication(false);
+            return filter;
+        }
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            // See https://docs.spring.io/spring-security/site/docs/current/reference/html5/#ns-custom-filters
+            // for list of filters and their order
+            http
+                .requestMatcher(EndpointRequest.to("jolokia")) // restrict this config to Jolokia
+                .csrf().disable()
+                .addFilter(identityHeaderAuthenticationFilter())
+                .addFilterBefore(getVerbIncludingAntiCsrfFilter(appProps, env), LogoutFilter.class)
+                .authorizeRequests()
+                    .requestMatchers(EndpointRequest.to("jolokia")).permitAll();
```

Comment:
See note above about /actuator/hawtio/jolokia...

Reply:
I just removed this bit.  The other security config handles access to all the actuator stuff.

Reply:
Okay, I updated this with a second commit.  (Feel free to squash merge if you like).

I've kept the same access control on the actuator endpoints as before (the endpoints are referenced by name and not by specific paths).  If we want to tighten it up we can, but my previous commit was too lenient with the other endpoints.  The Jolokia access is controlled in the config class specific to the Jolokia stuff.  We can require that people be authenticated there or belong to certain roles or anything similar.